packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -67,49 +71,70 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const [ homeId, setHomeId ] = useState();
+	useEffect( () => {
+		const effect = async () => {
+			try {
+				const { success, data } = await fetch(
+					addQueryArgs( '/', { '_wp-find-template': true } )
+				).then( ( res ) => res.json() );
+				if ( success ) {
+					let newHomeId = data.ID;
+					if ( newHomeId === null ) {
+						const { getEntityRecords } = resolveSelect( 'core' );
+						newHomeId = await getEntityRecords(
+							'postType',
+							'wp_template',
+							{
+								resolved: true,
+								slug: data.post_name,
+							}
+						)[ 0 ].id;
```

Comment:
You are awaiting the entire statement. That means this will always throw as `[0].id` doesn't exist in a promise.

You need to use parentheses to enforce the correct order of operations.

Reply:
:facepalm: 

Sorry, been distracted by maintenance duties for the past couple days, hence the sporadic/mediocre commits :confused:

Reply:
31ce3d1